library/alloc/src/vec/mod.rs

```diff
@@ -1617,7 +1617,28 @@ impl<T, A: Allocator> Vec<T, A> {
             return;
         }
 
-        /* INVARIANT: vec.len() > read >= write > write-1 >= 0 */
+        // Check if we ever want to remove anything.
+        // This allows to use copy_non_overlapping in next cycle.
+        // And avoids any memory writes if we don't need to remove anything.
+        let mut possible_remove_idx = 1;
+        let start = self.as_mut_ptr();
+        while possible_remove_idx < len {
+            let need_drop = unsafe {
```

Comment:
Nit: shouldn’t this have a safety comment?

Reply:
Added.